tt_metal/llrt/tt_cluster.cpp

```diff
@@ -346,30 +346,6 @@ int Cluster::get_device_aiclk(const chip_id_t &chip_id) const {
     return 0;
 }
 
-void Cluster::reset_debug_print_server_buffers() const {
-    for (const auto &[device_id, mmio_device_id] : this->device_to_mmio_device_) {
-        auto workers = get_soc_desc(device_id).workers;
```

Comment:
The issue that was causing hangs is from here: the `workers` vector used virtual core coords, but `write_core` requires physical core coords (it virtualizes them further down). This ultimately ended up with writes to harvested cores on n300 in some cases.

We already get a list of all physical cores in the print server, so moving this logic there and not using `workers` makes things work as intended

Reply:
I'm thinking we should seriously consider explicit typed coordinates now (virtual, physical, ... )

Reply:
We sort of have the abstraction layer called `metal_soc_desc`. we can probably make `get_soc_desc` unaccessible outside of the constructor of metal soc desc.